src/seedsigner/views/view.py

```diff
@@ -243,3 +243,50 @@ def run(self):
         ).display()
         
         return Destination(MainMenuView, clear_history=True)
+
+class MicroSDToastView(View):
+    def event_handler(self, action, mount_dir):
+        import time
+        
+        self.current_screen = self.renderer.canvas.copy()
+        
+        # Special case when screensaver is running
+        if self.controller.screensaver._is_running:
+            self.buttons.override_ind = True
+        
+        # MicroSDToastView blocks any attempts to use the Renderer in another thread so it
+        # never gives up the lock until it returns.
+        with self.renderer.lock:
```

Comment:
The `ToastOverlay` itself should claim the `renderer` lock and handle storing the `current_screen` and restoring it.

Related: the `views/screensaver.py` should really be in `screens/` since it's all UI rendering code and they all derive from `BaseScreen`. It being misplaced is just an artifact of the overall v0.5.0 organization becoming more clear after implementing it (and/or me being lazy at some point during the process).